sonar-db/src/main/java/org/sonar/db/user/GroupDao.java

```diff
@@ -44,19 +45,41 @@ public GroupDao(System2 system) {
     this.system = system;
   }
 
-  public GroupDto selectOrFailByName(DbSession session, String key) {
-    GroupDto group = selectByName(session, key);
+  /**
+   * @deprecated replaced by {@link #selectByName(DbSession, String, String)}
+   */
+  @Deprecated
+  public GroupDto selectOrFailByName(DbSession session, String name) {
+    GroupDto group = selectByName(session, name);
     if (group == null) {
-      throw new RowNotFoundException(String.format("Could not find a group with name '%s'", key));
+      throw new RowNotFoundException(String.format("Could not find a group with name '%s'", name));
     }
     return group;
   }
 
+  /**
+   * @deprecated replaced by {@link #selectByName(DbSession, String, String)}
+   */
+  @Deprecated
   @CheckForNull
   public GroupDto selectByName(DbSession session, String key) {
     return mapper(session).selectByKey(key);
   }
 
+  /**
+   * @param dbSession
+   * @param organizationUuid non-null UUID of organization (no support of "default" organization)
+   * @param name non-null group name
+   * @return the group with the given organization key and name
+   */
+  public Optional<GroupDto> selectByName(DbSession dbSession, String organizationUuid, String name) {
+    return Optional.ofNullable(mapper(dbSession).selectByName(organizationUuid, name));
+  }
+
+  /**
+   * @deprecated organization should be added as a parameter
+   */
+  @Deprecated
   public List<GroupDto> selectByNames(DbSession session, Collection<String> names) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)